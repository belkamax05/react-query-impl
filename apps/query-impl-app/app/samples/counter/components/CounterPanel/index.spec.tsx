import { render, screen, fireEvent } from '@testing-library/react';
import CounterPanel from '.';

describe('CounterPanel', () => {
  it('should render the counter value', () => {
    const counter = 5;
    const { baseElement, getByTestId } = render(<CounterPanel counter={counter} />);

    expect(getByTestId(`counter-value`)).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
  });

  it('should call onIncrement when the increment button is clicked', () => {
    const onIncrement = jest.fn();
    render(<CounterPanel counter={0} onIncrement={onIncrement} />);
    const incrementButton = screen.getByText('Inc');
    fireEvent.click(incrementButton);
    expect(onIncrement).toHaveBeenCalled();
  });

  it('should call onDecrement when the decrement button is clicked', () => {
    const onDecrement = jest.fn();
    render(<CounterPanel counter={0} onDecrement={onDecrement} />);
    const decrementButton = screen.getByText('Dec');
    fireEvent.click(decrementButton);
    expect(onDecrement).toHaveBeenCalled();
  });
});