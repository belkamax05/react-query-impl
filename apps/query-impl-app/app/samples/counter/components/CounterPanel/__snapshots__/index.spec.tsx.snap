// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterPanel should render the counter value 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-k008qs"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Inc
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiBox-root css-zg1vud"
      >
        <pre>
          Counter: 
          <strong
            data-testid="counter-value"
          >
            5
          </strong>
        </pre>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Dec
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</body>
`;
